Source/HelixToolkit.Wpf/Visual3Ds/Composite/ViewCubeVisual3D.cs

```diff
@@ -341,31 +341,30 @@ private void UpdateVisuals()
             var frontColor = Brushes.Red;
             var rightColor = Brushes.Green;
             var upColor = Brushes.Blue;
-            var up = this.ModelUpDirection;
-            var right = new Vector3D(0, 1, 0);
-            if (up.Z < 1)
+            var vecUp = this.ModelUpDirection;
+            var zIsUp = (vecUp.Z > 0) || (vecUp.Z < 0); // fix #4?
+            var vecLeft = new Vector3D(0, 1, 0);
+            if (!zIsUp)
             {
-                right = new Vector3D(0, 0, 1);
+                vecLeft = new Vector3D(0, 0, 1);
                 rightColor = Brushes.Blue;
                 upColor = Brushes.Green;
             }
 
-            var front = Vector3D.CrossProduct(right, up);
+            var vecFront = Vector3D.CrossProduct(vecLeft, vecUp);
 
-            this.AddCubeFace(front, up, frontColor, this.FrontText);
-            this.AddCubeFace(-front, up, frontColor, this.BackText);
-            this.AddCubeFace(right, up, rightColor, this.RightText);
-            this.AddCubeFace(-right, up, rightColor, this.LeftText);
-            this.AddCubeFace(up, right, upColor, this.TopText);
-            this.AddCubeFace(-up, -right, upColor, this.BottomText);
+            this.AddCubeFace(vecFront, vecUp, frontColor, this.FrontText);
+            this.AddCubeFace(-vecFront, vecUp, frontColor, this.BackText);
+            this.AddCubeFace(vecLeft, vecUp, rightColor, this.LeftText); // fix #183
+            this.AddCubeFace(-vecLeft, vecUp, rightColor, this.RightText);
+            this.AddCubeFace(vecUp, vecLeft, upColor, this.TopText);
+            this.AddCubeFace(-vecUp, -vecLeft, upColor, this.BottomText);
 
             var circle = new PieSliceVisual3D();
             circle.BeginEdit();
             circle.Center = (this.ModelUpDirection * (-this.Size / 2)).ToPoint3D();
             circle.Normal = this.ModelUpDirection;
-            circle.UpVector = this.ModelUpDirection.Equals(new Vector3D(0, 0, 1))
-                                  ? new Vector3D(0, 1, 0)
-                                  : new Vector3D(0, 0, 1);
+            circle.UpVector = vecLeft;

```

Comment:
This is just an optimisation. the previous code basically checked the UpDirection vector again and assigned (hard-coded) the same value that is in the `right` variable.
I'll add a comment.